js/ui/scheduler/workspaces/view_model/utils.js

```diff
@@ -25,3 +25,35 @@ export const getViewDataGeneratorByViewType = (viewType) => {
             return new ViewDataGenerator();
     }
 };
+
+export function alignToFirstDayOfWeek(date, firstDayOfWeek) {
+    const newDate = new Date(date);
+    let dayDiff = newDate.getDay() - firstDayOfWeek;
+
+    if(dayDiff < 0) {
+        dayDiff += 7;
+    }
+
+    newDate.setDate(newDate.getDate() - dayDiff);
+
+    return newDate;
+}
+
+export function alignToLastDayOfWeek(date, firstDayOfWeek) {
+    const newDate = alignToFirstDayOfWeek(date, firstDayOfWeek);
+    newDate.setDate(newDate.getDate() + 6);
+    return newDate;
+}
+
+export function calculateDaysBetweenDates(fromDate, toDate) {
+    const MS_IN_DAY = 24 * 60 * 60 * 1000;
+    return Math.round((toDate.getTime() - fromDate.getTime()) / MS_IN_DAY) + 1;
+}
```

Comment:
This implementation doesn't take into account a date's time.
So, the test case with (fromDate: '2023-06-28T23:59:00', toDate: '2023-06-29T00:01:00') returns 1,
but should return the 2 :)

I suggest the next implementation:
```suggestion
export const resetDateTime = (date) => {
    const dateCopy = new Date(date);
    dateCopy.setHours(0, 0, 0, 0);
    return dateCopy;
};

export const calculateDaysBetweenDates = (fromDate, toDate) => {
    const msDiff = resetDateTime(toDate).getTime() - resetDateTime(fromDate).getTime();
    // NOTE: We're adding the 1 here because:
    // (fromDate: 2023-01-01, toDate: 2023-01-01) => 1 day
    // (fromDate: 2023-01-01, toDate: 2023-01-03) => 3 days
    return Math.ceil(msDiff / MS_IN_DAY) + 1;
};
```